pkg/services/ngalert/image/service.go

```diff
@@ -73,36 +79,45 @@ func NewScreenshotImageServiceFromCfg(cfg *setting.Cfg, metrics prometheus.Regis
 	s = screenshot.NewSingleFlightScreenshotService(s)
 	s = screenshot.NewCachableScreenshotService(metrics, screenshotCacheTTL, s)
 	s = screenshot.NewObservableScreenshotService(metrics, s)
-
 	return &ScreenshotImageService{
 		store:       db,
 		screenshots: s,
 	}, nil
 }
 
-// NewImage returns a screenshot of the panel for the alert rule. It returns
-// ErrNoDashboard if the alert rule does not have a dashboard and ErrNoPanel
-// when the alert rule does not have a panel in a dashboard.
-func (s *ScreenshotImageService) NewImage(ctx context.Context, r *models.AlertRule) (*models.Image, error) {
+// NewImage returns a screenshot of the alert rule or an error.
+//
+// The alert rule must be associated with a dashboard panel for a screenshot to be
+// taken. If the alert rule does not have a Dashboard UID in its annotations,
+// or the dashboard does not exist, an ErrNoDashboard error is returned. If the
+// alert rule has a Dashboard UID and the dashboard exists, but does not have a
+// Panel ID in its annotations then an ErrNoPanel error is returned.
+func (s *ScreenshotImageService) NewImage(ctx context.Context, r *ngmodels.AlertRule) (*ngmodels.Image, error) {
```

Comment:
Is there a reason why we need rule as an argument when we use just two of its fields?

Reply:
Ok, but for that AlertRule model is also too much. If we choose to evaluate rules inside the image service, the condition and query should be enough. Ideally, the evaluation results that we keep in `state.State` should be enough to draw a chart (if not, we can put data frames that were returned by query evaluation and that contain all the points). 
I would rather go with a struct specific to the `image` package and convert the rule to it on the consumer side.